compiler/src/dotty/tools/backend/jvm/BCodeBodyBuilder.scala

```diff
@@ -315,20 +341,18 @@ trait BCodeBodyBuilder extends BCodeSkelBuilder {
             // but I was able to derrive it by reading
             // AbstractValidatingLambdaMetafactory.validateMetafactoryArgs
 
-            val Select(prefix, _) = fun
+            val DesugaredSelect(prefix, _) = fun
             genLoad(prefix)
           }
 
-          genLoadArguments(env, fun.symbol.info.paramTypes map toTypeKind)
+          genLoadArguments(env, fun.symbol.info.firstParamTypes map toTypeKind)
           generatedType = genInvokeDynamicLambda(NoSymbol, fun.symbol, env.size, functionalInterface)
 
         case app @ Apply(_, _) =>
           generatedType = genApply(app, expectedType)
 
-        case ApplyDynamic(qual, args) => sys.error("No invokedynamic support yet.")
-
         case This(qual) =>
-          val symIsModuleClass = tree.symbol.isModuleClass
+          val symIsModuleClass = tree.symbol.is(Flags.ModuleClass)
```

Comment:
Can we `import Flags._` to avoid having to select it everywhere?

Reply:
Done